### ISSUE TITLE
Import of azurerm_key_vault_key causes unnecessary replacement

### ISSUE THREAD

**Issue by fcatacut** [hashicorp/terraform-provider-azurerm#12357]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.14.11
+ provider registry.terraform.io/hashicorp/azurerm v2.64.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

### Affected Resource(s)

* `azurerm_key_vault_key`

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {
    key_vault {
      recover_soft_deleted_key_vaults = false
      purge_soft_delete_on_destroy    = true
    }
  }
}

data "azurerm_client_config" "current" {}

resource "random_string" "random" {
  length  = 8
  lower   = true
  special = false
  upper   = false
}

resource "azurerm_resource_group" "rg" {
  name     = "test-rg"
  location = "centralus"
}

resource "azurerm_key_vault" "kv" {
  name                = format("test-kv-%s", random_string.random.result)
  location            = azurerm_resource_group.rg.location
  resource_group_name = azurerm_resource_group.rg.name
  tenant_id           = data.azurerm_client_config.current.tenant_id

  sku_name                   = "standard"
  soft_delete_retention_days = 7

  access_policy {
    tenant_id = data.azurerm_client_config.current.tenant_id
    object_id = data.azurerm_client_config.current.object_id

    key_permissions = [
      "create",
      "get",
      "purge",
      "recover"
    ]

    secret_permissions = [
      "set",
    ]
  }
}

resource "azurerm_key_vault_key" "key" {
  name         = "test-key"
  key_vault_id = azurerm_key_vault.kv.id
  key_type     = "RSA"
  key_size     = 2048

  key_opts = [
    "decrypt",
    "encrypt",
    "sign",
    "unwrapKey",
    "verify",
    "wrapKey",
  ]
}
```

### Expected Behaviour

`terraform plan` should result in no changes assuming there are no configuration changes.

### Actual Behaviour

`terraform plan` indicates a "forces replacement"

```hcl
Terraform will perform the following actions:

  # azurerm_key_vault_key.key must be replaced
-/+ resource "azurerm_key_vault_key" "key" {
      + curve          = (known after apply)
      ~ e              = "AQAB" -> (known after apply)
      ~ id             = "https://test-kv-m4oloyv5.vault.azure.net/keys/test-key/2507a8c9a30e478bb4623215182eed52" -> (known after apply)
      + key_vault_id   = "/subscriptions/****/resourceGroups/test-rg/providers/Microsoft.KeyVault/vaults/test-kv-m4oloyv5" # forces replacement
      ~ n              = "..." -> (known after apply)
        name           = "test-key"
      - tags           = {} -> null
      ~ version        = "2507a8c9a30e478bb4623215182eed52" -> (known after apply)
      ~ versionless_id = "https://test-kv-m4oloyv5.vault.azure.net/keys/test-key" -> (known after apply)
      + x              = (known after apply)
      + y              = (known after apply)
        # (3 unchanged attributes hidden)

      - timeouts {}
    }

Plan: 1 to add, 0 to change, 1 to destroy.

```

### Steps to Reproduce

1. Run `terraform apply`.
2. Remove the resource from the state file: `terraform state rm 'azurerm_key_vault_key.key'`.
3. (Optional) Run `terraform plan` to confirm that the KV Key will be added.
4. (Optional) Run `terraform apply` which should fail with a "resource with the ID already exists".
5. Import the resource: `terraform import 'azurerm_key_vault_key.key' '<kv key id>'`.
6. Run `terraform plan`.

### Important Factoid / Workaround

It looks as if the `"key_vault_id"` value isn't getting set in the state file. Once I manually updated it from `null` to the Key Vault ID, `terraform plan` show no changes.

**Comment by mikemonteith** [comment]
I have come across this bug. The "key_vault_id" created by `terraform import` is missing a trailing slash. When you run `terraform apply`, the calculated key_vault_id doesn't match and so it forces replacement.

I manually edited the state by adding a trailing slash which fixed my problem.

It's interesting to hear that @fcatacut had key_vault_id set to `null`. My terraform versions are:
```
Terraform v1.0.11
on darwin_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.84.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
```

**Comment by mikemonteith** [comment]
After investigating a bit more, this only happens when the keyvault iteslf has also been imported with `terraform import` into the state, *and the resource id used for importing had a trailing slash*.

e.g
```
#good
terraform import azurerm_key_vault.key_vault /subscriptions/07748954-52d6-46ce-95e6-2701bfc715b4/resourceGroups/terraform-debug-test/providers/Microsoft.KeyVault/vaults/terraform-debug-test

#bad
terraform import azurerm_key_vault.key_vault /subscriptions/07748954-52d6-46ce-95e6-2701bfc715b4/resourceGroups/terraform-debug-test/providers/Microsoft.KeyVault/vaults/terraform-debug-test/
```

It would be nice if terraform removed the trailing slash for us in this situation!

**Comment by mikemonteith** [comment]
Having read the original issue again and investigated some more. I've opened a new issue for my trailing slash problem. I think it's slightly different from the original issue.
https://github.com/hashicorp/terraform-provider-azurerm/issues/14325

**Comment by bcreynolds** [comment]
I'm seeing the same original issue as @fcatacut -- nothing to do with trailing slashes. terraform import of an "azurerm_key_vault_key" resource sets the "key_vault_id" value to null in the state, forcing a key replacement on the next apply. 

```
Terraform v1.1.2
on windows_amd64
+ provider registry.terraform.io/hashicorp/azurerm v2.90.0
```

**Comment by logico-philosophicus** [comment]
I am seeing same issue - key vault id has null value in state after importing an azurerm_key_vault_key

terraform v1.1.4
azurerm v2.82.0

**Comment by aledeniz** [comment]
I am seeing same issue: the key vault id has null value in state after importing an azurerm_key_vault_key.
I worked that around manually editing the state file.
terraform v1.1.5
azurerm v2.80.0

**Comment by Noel-Jones** [comment]
Have created a PR to handle this in the same manner ar for secrets and certificates.

**Comment by katbyte** [comment]
fixed by  #15670

**Comment by logico-philosophicus** [comment]
Still seeing this issue:

terraform v1.1.7
azurerm v2.98.0
